### PR TITLE
Add font scaling client convar & settings page

### LINKED ISSUE #30: Add font scaling client convar (50% -> 200% font size slider)
- [ ] Add a `advisor_text_scale` convar, takes a number between 0.5 and 2
- [ ] Create a `Advisor.NumSlider` UI element with the same style as the others. Derives from `DNumSlider`
- [ ] Create the `Settings` menu page. For now, only add the text scale setting.
- [ ] Call the font update function in `cl_fonts` when the value is changed.

This will allow users to scale the font according to their screen's pixel density. Resolution scaling alone won't give us a "perfect" result.

### PR DESCRIPTION
**Pull request by Guthen**
Resolves #30 

### Result
Result

### Change List
+ Add VGUI `Advisor.NumSlider`, `Advisor.Slider` & `Advisor.Menu.Settings`
+ Add ConVar `advisor_text_scale` accepting value in the range from `0.5` to `2`
+ Add `Advisor.Theme.Slider` theme
+ Add Static Font `Advisor:Rubik.StaticSmall` for `Advisor.Slider` notch's text
+ Add Min Width to `Advisor.ScrollPanel`
+ Add DockPadding to `Advisor.Button`
+ Move `Advisor.Menu/PopulateOptions` function to `Advisor.ScrollPanel/PopulateOptions`
+ Add suffix to `Advisor.TextEntry`
+ Fix responsive due to text scaling on `Advisor.TitleBar`, `Advisor.MenuOption`, `Advisor.MenuCategory`, `Advisor.HeaderBox`, `Advisor.Button`

### CODE REVIEW

File: lua/advisor-modules/ui/vgui/cl_advisor_numslider.lua [hunk 1]
```diff
@@ -1,3 +1,92 @@
 local PANEL = {}
 
+function PANEL:Init()
+    local parent = self
+    
+    -- Replacing some GUI with ours (https://github.com/Facepunch/garrysmod/blob/master/garrysmod/lua/vgui/dnumslider.lua)
+    
+    -- Replace DSlider with Advisor.Slider
+    self.Slider:Remove()
```
Comment (Erlite) [inline lua/advisor-modules/ui/vgui/cl_advisor_numslider.lua:9]: Please fix the weird indentation in this function.

**Review by Erlite — CHANGES_REQUESTED** [review]
Just a quick fix on the indentation and it'll be good to merge.

File: lua/advisor-modules/ui/vgui/cl_advisor_numslider.lua (hunk 1 above)
Comment (Guthen) [reply lua/advisor-modules/ui/vgui/cl_advisor_numslider.lua:9]: Done!

**Review by Erlite — APPROVED** [review]
lgtm

**Pull request merged**